Controllers/FrontController.php

```diff
@@ -1,44 +1,132 @@
 <?php
+declare(strict_types=1);
 
 namespace Controllers;
 
+use Interfaces\IDataManagement;
+use Models\DataRegistry;
+use Models\Logger;
+use Models\Server;
+use Models\Session;
+use Models\Post;
+use Models\File;
+use Models\Config;
+
+/**
+ * Main class for routing
+ *
+ * @package Controllers
+ */
 class FrontController
 {
-    protected $controller;
-    protected $action, $params;
-    public static $instance;
+    /**
+     * Given controller
+     */
+    protected string $controller;
+
+    /**
+     * Given action
+     */
+    protected string $action;
+
+    /**
+     * Given params values
+     */
+    protected ?array $params = null;
+
+    /**
+     * Request uri
+     */
+    protected string $request;
+
+    /**
+     * Object for access to server data
+     */
+    private IDataManagement $serverData;
 
-    public static function getInstance()
+    public Logger $logger;
+
+
+    public function __construct()
     {
-        if (!(self::$instance instanceof self)) {
-            self::$instance = new self();
+        $this->logger = new Logger();
+        try {
+            $this->registerData();
+        } catch (\Exception $exception) {
+            $this->logger->log($exception->getMessage() . $exception->getTraceAsString());

```

Comment:
why do you need to have two try catch conditions here? i guess you can use only one here because the only thing you do is just logging action. 
```
try {
            $this->registerData();
            $this->serverData = DataRegistry::getInstance()->get('server');
        } catch (\Exception $exception) {


```